### PR TITLE
[feat]: [PLG-2132]: added admin access control check for free, trial and extend trial license APIs

### PR DESCRIPTION
**Pull request by kapil-harness**
## Describe your changes
added admin access control check for free, trial and extend trial license APIs

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/46284)

### CODE REVIEW

File: 930-ng-license-manager/src/main/java/io/harness/licensing/api/resource/LicenseResource.java [hunk 1]
```diff
@@ -198,7 +199,7 @@ public LicenseResource(LicenseService licenseService) {
         @io.swagger.v3.oas.annotations.responses.
         ApiResponse(responseCode = "default", description = "Returns the Free License of the specified Module.")
       })
-  @NGAccessControlCheck(resourceType = ResourceTypes.LICENSE, permission = VIEW_LICENSE_PERMISSION)
```
Comment (Xin-Harness) [inline 930-ng-license-manager/src/main/java/io/harness/licensing/api/resource/LicenseResource.java:201]: Any specific reason that the permission needs to be changed to edit license permission?

Comment (Xin-Harness) [reply 930-ng-license-manager/src/main/java/io/harness/licensing/api/resource/LicenseResource.java:201]: Just a reminder, in complete signup service step, it checks rbac setup completion, which checks View license permission. If the license permission get change in API, the complete signup service step may also need to get changed.

**Review by thomas-grasman — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46284)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by kapil-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46284)* status: 1 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`930-ng-license-manager/src/main/java/io/harness/licensing/api/resource/LicenseResource.java` line 0 at r1](https://reviewable.io/reviews/harness/harness-core/46284#gh-1161732162:-NSg6rmu15vuz_VbzOpN:bu7o87) ([raw file](https://github.com/harness/harness-core/blob/906841822ff3ffcab9f5413c8b649c01cfc568f1/930-ng-license-manager/src/main/java/io/harness/licensing/api/resource/LicenseResource.java#L0)):*
<details><summary><i>Previously, Xin-Harness (Xin Shao) wrote…</i></summary><blockquote>

Just a reminder, in complete signup service step, it checks rbac setup completion, which checks View license permission. If the license permission get change in API, the complete signup service step may also need to get changed.
</blockquote></details>

We want only the account admins to be able to access start free/trial licensing from subscriptions section.

so my changes are for license APIs and not signup. Above code is for signup flow and for license APIs we don’t have any such rbacsetup.

**Review by Xin-Harness — APPROVED** [review]

**Review by neelam-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46284)* status: 1 file(s) to review. 1 change requests. 1. unresolved discussions. 

___
*[`930-ng-license-manager/src/main/java/io/harness/licensing/api/resource/LicenseResource.java` line 202 at r1](https://reviewable.io/reviews/harness/harness-core/46284#-NSgLHhe0gHk_GxtPz0F:-NSgLHhe0gHk_GxtPz0G:bmhknir) ([raw file](https://github.com/harness/harness-core/blob/906841822ff3ffcab9f5413c8b649c01cfc568f1/930-ng-license-manager/src/main/java/io/harness/licensing/api/resource/LicenseResource.java#L202)):*
> ```Java
>         ApiResponse(responseCode = "default", description = "Returns the Free License of the specified Module.")
>       })
>   @NGAccessControlCheck(resourceType = ResourceTypes.LICENSE, permission = EDIT_LICENSE_PERMISSION)
> ```

How did we test these changes ?

**Review by kapil-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46284)* status: 1 file(s) to review. 1 change requests. 2. unresolved discussions. 

___
*[`930-ng-license-manager/src/main/java/io/harness/licensing/api/resource/LicenseResource.java` line 202 at r1](https://reviewable.io/reviews/harness/harness-core/46284#-NSgLHhe0gHk_GxtPz0F:-NSgbxGH0iQNXIob0fQr:b4agi3r) ([raw file](https://github.com/harness/harness-core/blob/906841822ff3ffcab9f5413c8b649c01cfc568f1/930-ng-license-manager/src/main/java/io/harness/licensing/api/resource/LicenseResource.java#L202)):*
<details><summary><i>Previously, neelam-harness wrote…</i></summary><blockquote>

How did we test these changes ?
</blockquote></details>

We have not tested these changes locally because we are using an already existing admin permission and we know for a fact that APIs annotated with this permission can only be used by an admin. 

EDIT_LICENSE_PERMISSION is used by subscription APIs(PFA).

**Review by neelam-harness — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46284)* status: 1 file(s) to review. 1. unresolved discussions. 

___
*[`930-ng-license-manager/src/main/java/io/harness/licensing/api/resource/LicenseResource.java` line 202 at r1](https://reviewable.io/reviews/harness/harness-core/46284#-NSgLHhe0gHk_GxtPz0F:-NSgeCSaAcwMrR3zqv2i:b-cyv2hx) ([raw file](https://github.com/harness/harness-core/blob/906841822ff3ffcab9f5413c8b649c01cfc568f1/930-ng-license-manager/src/main/java/io/harness/licensing/api/resource/LicenseResource.java#L202)):*
<details><summary><i>Previously, kapil-harness (Kapil Choudhary) wrote…</i></summary><blockquote>

We have not tested these changes locally because we are using an already existing admin permission and we know for a fact that APIs annotated with this permission can only be used by an admin. 

EDIT_LICENSE_PERMISSION is used by subscription APIs(PFA).


</blockquote></details>

Thanks for clarifying.

**Pull request merged**